decidim-core/app/services/decidim/settings_change.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module Decidim
+  # This is a helper class in order to publish feature and settings changes
+  # so that components can react to these changes and send notifications to users.
+  class SettingsChange
+    # Publishes a change to ActiveSupport::Notifications.
+    #
+    # feature - The Decidim::Feature where the changes have been applied.
+    # previous_settings - A Hash or a Decidim::SettingsManifest schema with the settings before changing them.
+    # current_settings - A Hash or a Decidim::SettingsManifest schema with the current settings.
+    # previous_step - The optional ParticipatoryProcessStep that was previously active.
+    # current_step - The optional ParticipatoryProcessStep that is currently active.
+    def self.publish(feature, previous_settings, current_settings, previous_step = nil, current_step = nil)
```

Comment:
This is coupled to `previous_step` and `current_step` - shouldn't be if it belongs to `core`. What's the reason to pass that in?

Reply:
I actually thought about this too, I'm not sure there's a reason. I could remove it and if I need it in the future re-add it.

Reply:
My point is that this is a leak - only participatory processes know about steps. Let's remove it and NEVAR PUT IT HERE AGAIN 👍